File: scripts/R/03_advanced/read_many_duckdb_files.R
library(duckdb)

# Create an in-memory connection
con <- dbConnect(duckdb::duckdb())

# Attach multiple databases wuth the same table structure
# If you have many files, you can dynamically attach only the relevant ones
dbExecute(con, "ATTACH 'data_2023.duckdb' AS db2023")
dbExecute(con, "ATTACH 'data_2024.duckdb' AS db2024")

# Query across them
result <- dbGetQuery(con, "
  SELECT col1, col2 FROM db2023.my_table
  UNION ALL
  SELECT col1, col2 FROM db2024.my_table
  WHERE some_column = 'value'
")

dbDisconnect(con, shutdown = TRUE)
print(result)

File: scripts/Python/03_advanced/read_many_duckdb_files.py
import duckdb
import pandas as pd

# Create an in-memory DuckDB connection
con = duckdb.connect()

# Attach multiple DuckDB databases
con.execute("ATTACH 'data_2023.duckdb' AS db2023")
con.execute("ATTACH 'data_2024.duckdb' AS db2024")

# Query across them
result = con.execute("""
    SELECT col1, col2 FROM db2023.my_table
    UNION ALL
    SELECT col1, col2 FROM db2024.my_table
    WHERE some_column = 'value'
""").fetchdf()

# Close the connection
con.close()

print(result)
